pkg/services/mysql/dbms_arm_template.go

```diff
@@ -9,61 +9,39 @@ var dbmsARMTemplateBytes = []byte(`
 			"location": {
 				"type": "string"
 			},
-			"administratorLoginPassword": {
-				"type": "securestring"
-			},
-			"serverName": {
-				"type": "string",
-				"minLength": 2,
-				"maxLength": 63
-			},
-			"skuName": {
-				"type": "string",
-				"allowedValues": [ "MYSQLB50", "MYSQLB100", "MYSQLS100","MYSQLS200", "MYSQLS400", "MYSQLS800" ]
-			},
-			"skuTier": {
-				"type": "string",
-				"allowedValues": [ "Basic", "Standard"]
-			},
-			"skuCapacityDTU": {
-				"type": "int",
-				"allowedValues": [ 50, 100, 200, 400, 800 ]
-			},
-			"skuSizeMB": {
-				"type": "int",
-				"minValue": 51200,
-				"maxValue": 128000
-			},
-			"sslEnforcement": {
-				"type": "string",
-				"allowedValues": [ "Enabled", "Disabled" ],
-				"defaultValue": "Enabled"
-			},
 			"tags": {
 				"type": "object"
 			}
 		},
 		"variables": {
-			"DBforMySQLapiVersion": "2017-04-30-preview"
+			"DBforMySQLapiVersion": "2017-12-01",
+			"ServerName" : "{{ .serverName }}"
```

Comment:
This doesn't buy us anything, afaict. Might be easier to read if we just use `{{ .serverName }}` wherever.

Reply:
It’s used within the firewall block, which is selecting the .firewallRules so you can’t use {{ .serviceName }} directly. Seemed more readable to just declare an arm variable and use it throughout, but open to suggestions!

Reply:
@jeremyrickard in cases where `{{range}}` affects what `.` references, you can still reference what `.` referenced outside that loop by using `$.`

i.e. `$.serverName`

Admittedly, I didn't know this until quite recently.

I'm going to amend the PR with this fix.